docs/internal-architecture.md

```diff
@@ -0,0 +1,91 @@
+---
+description: Internal architecture of the project.
+---
+
+# Internal architecture
+
+### Trackers
+
+A tracker is usually in the type of an index tracker for a coin or token on a blockchain network.
+
+Index trackers take placed in the `pkg/tracker` package of the Telliot repository. the index tracker struct is as follow:
+
+```go
+type IndexTracker struct {
+	DB               db.DataServerProxy
+	Name             string
+	Identifier       string
+	Symbols          []string
+	Source           DataSource
+	Interval         time.Duration
+	Param            string
+	Type             IndexType
+	lastRunTimestamp time.Time
+}
+```
+
+an IndexTracker struct is related to an index object from the `indexes.json` file. based on `Source` there are two main types of trackers: HTTP trackers, on-chain trackers.
+
+#### HTTP trackers
+
+An index tracker that fetches and parses tracker info using an HTTP call. the `indexes.json` entry for such trackers could be like this:
+
+```javascript
+    "AMPL/BTC": [
+        {
+            "URL": "https://api-pub.bitfinex.com/v2/tickers?symbols=tAMPBTC",
+            "param": "$[0][7,8]"
```

Comment:
Also add an `interval` as an example to show that it can be set.
Also add `parser` and mention what id defaults to when not added. 
For the params explain that these are additional parameters for the parser and for the jsonpath parser it is the jsonpath param on how to parse the output, with maybe a link to the jsonpath website.